google/cloud/storage/benchmarks/storage_throughput_benchmark.cc

```diff
@@ -280,9 +288,11 @@ std::string CreateChunk(google::cloud::internal::DefaultPRNG& generator,
   std::string const random_data =
       gcs_bm::MakeRandomData(generator, gcs_bm::kMiB);
   std::string chunk;
+  chunk.reserve(desired_size);
```

Comment:
The `reserve()` doesn't really help if the `append()` exceeds it.

Reply:
I think it still "helps", it reduces the number of allocations (and copies) from `log(desired_size / kMiB)` to **2** (the worst case you mention) or **1** (if `desired_size % kMiB == 0`, which I think it is most of the time) .

Reply:
> reduces the number of allocations (and copies)

The number of copy calls, yes ... but not the amount of data copied.  The "2" case still copies O(n) data, which, I'd say, dominates the O(log n) allocations.

Not a big deal either way of course, and particularly so if "1" is the expected case.